Bug FIx: PCI logging

jinhoyrivos opened the pull request:
- arrayvec uses sdt (not checked by #![cfg_attr(not(test), no_std)])
- re-implemented logging wihout arrayvec

Bug=[SW-8085]

furquan-rivos commented on src/lib.rs:
```diff
@@ -4,9 +4,6 @@
 
 #![cfg_attr(not(test), no_std)]
 
-use arrayvec::ArrayString;
```
Does it not work to disable the std features in arrayvec i.e. make it no_std compatible?

furquan-rivos commented on Cargo.toml:
```diff
@@ -10,7 +10,7 @@ edition = "2021"
 [dependencies]
 # implicitly defines a feature like "dep:log"
 log = { version = "0.4.14", default-features = false, optional = true }
-arrayvec = "*"
```
Commit message needs update.

furquan-rivos commented on src/lib.rs:
```diff
@@ -1074,7 +1074,7 @@ impl Device {
 
     fn print_config_space_bytes(&self, num_bytes: usize) {
-        const BUFFER_SIZE: usize = 100;
+        const BUFFER_SIZE: usize = 70;
```
unrelated change?

jinhoyrivos replied:
I didn't see this reviews, sorry. I updated the commit msg and made separate commits for the changed buffer sizes

jinhoyrivos committed:
print_config_space_bytes@lib.rs: Buffer size changed

- updated to initialize smaller buffer space.

jinhoyrivos committed:
Bug FIx: PCI logging

- arrayvec uses the standard library (std) and is not gated by #![cfg_attr(not(test), no_std)]
- Updated to disable the default-features for arrayvec to prevent usage of the standard library (std)

Bug=[SW-8085]
